src/box/memtx_space.c

```diff
@@ -293,6 +299,9 @@ memtx_space_replace_all_keys(struct space *space, struct tuple *old_tuple,
 			     enum dup_replace_mode mode,
 			     struct tuple **result)
 {
+	if (memtx_space_on_replace(space) != 0)
```

Comment:
Is it needed? AFAIU `memtx_space_replace_all_keys` is used only after recover completion.

Reply:
Also the `memtx_space_replace_all_keys` is [used](https://github.com/tarantool/tarantool/blob/584c3e00c410fc731dacfc649762060717c85f6c/src/box/memtx_space.c#L1144) used for all system spaces. But they have enabled all indexes from the beginning, so no SK nor PK build required after they're recovered from the snapshot.

Also the `memtx_space_replace_primary_key` callback is only used after the initial recovery is complete, so there's no reason to perform the logic in it too.

Removed the call from both `memtx_space_replace_all_keys` and `memtx_space_replace_primary_key`, it's only in the `memtx_space_replace_build_next` now.